src/api/procedures/deletePortfolio.ts

```diff
@@ -0,0 +1,71 @@
+import BigNumber from 'bignumber.js';
+
+import { NumberedPortfolio } from '~/api/entities';
+import { PolymeshError, Procedure } from '~/base';
+import { ErrorCode } from '~/types';
+import { numberToU64, stringToIdentityId } from '~/utils';
+
+export interface DeletePortfolioParams {
+  did: string;
+  id: BigNumber;
+}
+
+/**
+ * @hidden
+ */
+export async function prepareDeletePortfolio(
+  this: Procedure<DeletePortfolioParams>,
+  args: DeletePortfolioParams
+): Promise<void> {
+  const {
+    context: {
+      polymeshApi: {
+        query: { portfolio: queryPortfolio },
+        tx: { portfolio },
+      },
+    },
+    context,
+  } = this;
+
+  const { did, id } = args;
+
+  const numberedPortfolio = new NumberedPortfolio({ did, id }, context);
+  const identityId = stringToIdentityId(did, context);
+  const rawPortfolioNumber = numberToU64(id, context);
+
+  const [isOwned, rawPortfolioName, portfolioBalances] = await Promise.all([
+    numberedPortfolio.isOwned(),
+    queryPortfolio.portfolios(identityId, rawPortfolioNumber),
+    numberedPortfolio.getTokenBalances(),
+  ]);
+
+  if (rawPortfolioName.isEmpty) {
+    throw new PolymeshError({
+      code: ErrorCode.ValidationError,
+      message: "The Portfolio doesn't exist",
+    });
+  }
+
+  if (!isOwned) {
+    throw new PolymeshError({
+      code: ErrorCode.ValidationError,
+      message: 'You are not the owner of this Portfolio',
+    });
+  }
+
+  if (portfolioBalances.length > 0) {
+    if (portfolioBalances.every(balance => !balance.total.eq(0))) {
```

Comment:
This will return true if every balance is different than 0. What you need to check is if at least one balance is greater than 0. Also, you can destructure balance to get the total

` if (!portfolioBalances.every(({ total }) => total.eq(0)))` or `if (portfolioBalances.some(({ total }) => total.gte(0)))`